### PR TITLE
Grd 398

### PR DESCRIPTION
**Pull request by lheisler**

### CODE REVIEW

File: rnaSeqQC.wdl [hunk 1]
```diff
@@ -339,35 +367,23 @@ task picard {
...
-    # Environment variables from modulefiles:
-    # $PICARD_ROOT <- picard
-    # $HG38_ROOT <- hg38
-    # $HG38_REFFLAT_ROOT <- hg38-refflat
-
-    # VALIDATION_STRINGENCY=SILENT prevents BAM parsing errors with the given REFERENCE_SEQUENCE
-
-    # check if HG19_ROOT or HG38_ROOT variable is set by environment module
   command <<<
+    samtools sort ~{bamFile} > sorted.bam
```
Comment (pruzanov) [inline rnaSeqQC.wdl:381]: since it is a multi-step task now, it would be wise to add `set -euo pipefail` on top

**Review by pruzanov — APPROVED** [review]
I approved but there may be one of my comments which suggests a change

File: rnaSeqQC.wdl [hunk 2]
```diff
@@ -384,8 +400,81 @@ task picard {
...
   meta {
 	  output_meta: {
-      result: "Text file with Picard output"
+      result: "Text file with Picard InsertSize metrics output"
 	  }
   }
+  }
+  
+  
+  task picard {
```
[the remark this answers is not in the document]
Comment (lheisler) [reply rnaSeqQC.wdl:409]: this task is still in use as it was used before

File: rnaSeqQC.wdl [hunk 3]
```diff
@@ -339,35 +367,23 @@ task picard {
...
-
-    # VALIDATION_STRINGENCY=SILENT prevents BAM parsing errors with the given REFERENCE_SEQUENCE
-
-    # check if HG19_ROOT or HG38_ROOT variable is set by environment module
   command <<<
+    samtools sort ~{bamFile} > sorted.bam
     java -Xmx~{picardMem}M \
-    -jar $PICARD_ROOT/picard.jar CollectRnaSeqMetrics \
-    I=~{bamFile} \
+    -jar $PICARD_ROOT/picard.jar CollectInsertSizeMetrics \
```
[the remark this answers is not in the document]
Comment (lheisler) [reply rnaSeqQC.wdl:383]: the original picard CollectRnaSeqMetrics is maintained in the wdl, and is still processing the genomic bam file.  as discussed elsewhere, this does NOT calculate insert sizes.  I am not sure what metrics from there are being reported elsewhere

there is a new task that i wrote picardInsertSize, which can take the transcriptome bam file, but doesn't actually get called.  I could remove it; i noted that it wasn't getting called but thought to keep it in there should we find a need to use it.
Instead, the insert size is being called by the bamqc_transcriptome task (as well as other bamqc metrics)

File: rnaSeqQC.wdl (hunk 1 above)
Comment (lheisler) [reply rnaSeqQC.wdl:383]: done

**Review by iainrb — APPROVED** [review]
Looks good to me, approved

**Pull request merged**